CRM/Core/Error.php

```diff
@@ -634,19 +637,29 @@ public static function debug_query_result($query) {
   /**
    * Obtain a reference to the error log.
    *
-   * @param string $comp
+   * @param string $prefix
    *
    * @return Log
    */
-  public static function createDebugLogger($comp = '') {
-    if (!isset(\Civi::$statics[__CLASS__]['logger_file' . $comp])) {
+  public static function createDebugLogger($prefix = '') {
+    self::generateLogFileName($prefix);
+    return Log::singleton('file', \Civi::$statics[__CLASS__]['logger_file' . $prefix], '');
```

Comment:
Nitpick: a reader looking at those two lines probably wouldn't understand that `generateLogFileName()` has the side-effect of modifying `\Civi::$statics[__CLASS__]['logger_file' . $prefix]` -- it would be more expected that `generateLogFileName()` would actually return the file name.